src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java

```diff
@@ -902,104 +920,72 @@ public void logOut(String wikiURL, String articleName) {
-//      httpPost4.setEntity(new UrlEncodedFormEntity(nvps2,
-//                                                   StandardCharsets.UTF_8));
-//
-//      response = httpclient.execute(httpPost4);
-//
-////      entity = response.getEntity();
-////      for (int i = 0; i < 10; i++) {
-////
-////
-////        xmlResponse = EntityUtils.toString(entity);
-////
-////        xmlResponseArr = xmlResponse.split("\"");
-////
-////        if (xmlResponse.contains("WrongPass")) {
-////          throw new WebDriverException("Incorrect password provided for user: " + userName);
-////        }
-////
-////        if (xmlResponseArr.length >= 11) {
-////          break;
-////        }
-////      }
-//
-//      PageObjectLogging.log("LOGIN HEADERS: ", response.toString(), true);
-////      PageObjectLogging.log("LOGIN RESPONSE: ", xmlResponse, true);
-//
-//      for (org.apache.http.cookie.Cookie cookie : cookieStore.getCookies()) {
-//        System.out.println(cookie.toString());
-//        driver.manage().addCookie(
-//            new Cookie(cookie.getName(), cookie.getValue(), cookie.getDomain(), cookie.getPath(),
-//                       cookie.getExpiryDate(), false));
-//      }
-//
-//      try {
-//        driver.get(wikiURL);
-//      } catch (TimeoutException e) {
-//        PageObjectLogging.log("loginCookie",
-//                              "page timeout after login by cookie", true);
-//      }
-//
+    String client_id = HeliosConfig.getClientId();
+    String client_secret = HeliosConfig.getClientSecret();
+    String heliosBaseUrl = HeliosConfig.getUrl(HeliosConfig.HeliosController.TOKEN);
+
+    try {
+      CookieStore cookieStore = new BasicCookieStore();
+
+      HttpClient httpclient = HttpClientBuilder.create()
+          .setConnectionBackoffStrategy(new DefaultBackoffStrategy())
+          .setDefaultCookieStore(cookieStore)
+          .disableAutomaticRetries()
+          .build();
+      HttpPost httpPost = new HttpPost(heliosBaseUrl);
+      List<NameValuePair> nvps = new ArrayList<NameValuePair>();
+
+      nvps.add(new BasicNameValuePair("grant_type", HeliosConfig.GrantType.PASSWORD.getGrantType()));
+      nvps.add(new BasicNameValuePair("client_id", client_id));
+      nvps.add(new BasicNameValuePair("client_secret", client_secret));
+      nvps.add(new BasicNameValuePair("username", userName));
+      nvps.add(new BasicNameValuePair("password", password));
+
+      httpPost.setEntity(new UrlEncodedFormEntity(nvps, StandardCharsets.UTF_8));
+
+      HttpResponse response = httpclient.execute(httpPost);
+      HttpEntity entity = response.getEntity();
+
+      JSONObject responseValue = new JSONObject(EntityUtils.toString(entity));
+
+      PageObjectLogging.log("LOGIN HEADERS: ", response.toString(), true);
+      PageObjectLogging.log("LOGIN RESPONSE: ", responseValue.toString(), true);
+
+      String token = responseValue.getString("access_token");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "access_token" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1192)